### PR TITLE
Phase-1: Migrate QBO to platform ( Expenses API )

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/fyle/helpers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+from apps.workspaces.models import FyleCredential
+
+from .utils import FyleConnector
+
+logger = logging.getLogger(__name__)
+logger.level = logging.INFO
+
+
+def compare_tpa_and_platform_expenses(tpa_expenses: List[dict], platform_expenses: List[dict], workspace_id: int) -> None:
```
Comment (Shwetabhk) [inline apps/fyle/helpers.py:14]: Add a to do to remove this later

File: fyle_qbo_api/settings.py [hunk 2]
```diff
@@ -172,6 +172,10 @@
             'handlers': ['debug_logs'],
             'propagate': True,
         },
+        'fyle_integrations_platform_connector': {
```
Comment (Shwetabhk) [inline fyle_qbo_api/settings.py:175]: Add a to do to remove this later

File: apps/fyle/tasks.py [hunk 3]
```diff
@@ -75,24 +85,36 @@ def async_create_expense_groups(workspace_id: int, fund_source: List[str], task_
...
+
+            source_account_type = []
+            for source in fund_source:
+                source_account_type.append(SOURCE_ACCOUNT_MAP[source])
+
+            expenses = platform.expenses.get(
+                source_account_type, expense_group_settings.expense_state, last_synced_at, True
             )
 
+            compare_tpa_and_platform_expenses(tpa_expenses, expenses, workspace_id)
```
Comment (Shwetabhk) [inline apps/fyle/tasks.py:116]: Add a to do to clean this function up later

**Review by Shwetabhk — APPROVED** [review]

File: apps/fyle/models.py [hunk 4]
```diff
@@ -81,56 +94,39 @@ def create_expense_objects(expenses: List[Dict], workspace_id: int):
...
+                    'project': expense['project'],
                     'expense_number': expense['expense_number'],
+                    'org_id': expense['org_id'],
                     'claim_number': expense['claim_number'],
                     'amount': expense['amount'],
                     'currency': expense['currency'],
                     'foreign_amount': expense['foreign_amount'],
                     'foreign_currency': expense['foreign_currency'],
-                    'tax_amount': expense['tax'],
-                    'tax_group_id': expense['tax_group_id'],
```
Comment (NileshPant1999) [inline apps/fyle/models.py:115]: @Sravanksk why you are removing these ??

File: apps/fyle/models.py [hunk 5]
```diff
@@ -81,56 +94,39 @@ def create_expense_objects(expenses: List[Dict], workspace_id: int):
...
                     'settlement_id': expense['settlement_id'],
                     'reimbursable': expense['reimbursable'],
-                    'billable': expense['billable'],
```
Comment (NileshPant1999) [inline apps/fyle/models.py:118]: this too

File: apps/mappings/tasks.py [hunk 6]
```diff
@@ -88,7 +90,7 @@ def auto_create_tax_codes_mappings(workspace_id: int):
     try:
         fyle_credentials: FyleCredential = FyleCredential.objects.get(workspace_id=workspace_id)
 
-        fyle_connection = FylePlatformConnector(
+        fyle_connection = PlatformConnector(
             refresh_token=fyle_credentials.refresh_token,
             workspace_id=workspace_id
```
Comment (NileshPant1999) [inline apps/mappings/tasks.py:95]: @Sravanksk the parameters are wrong !!

File: apps/fyle/models.py [hunk 7]
```diff
@@ -81,56 +94,42 @@ def create_expense_objects(expenses: List[Dict], workspace_id: int):
...
-                    'category': expense['category_name'],
+                    'category': expense['category'],
                     'sub_category': expense['sub_category'],
-                    'project': expense['project_name'],
+                    'project': expense['project'],
                     'expense_number': expense['expense_number'],
+                    'org_id': expense['org_id'],
                     'claim_number': expense['claim_number'],
-                    'amount': expense['amount'],
+                    'amount': round(expense['amount'], 2),
```
Comment (ashwin1111) [inline apps/fyle/models.py:108]: Why are we doing this additionally now?

File: apps/fyle/tasks.py [hunk 8]
```diff
@@ -75,22 +84,20 @@ def async_create_expense_groups(workspace_id: int, fund_source: List[str], task_
                     'gte:{0}'.format(datetime.strftime(last_synced_at, '%Y-%m-%dT%H:%M:%S.000Z'))
                 )
 
-            fyle_credentials = FyleCredential.objects.get(workspace_id=workspace_id)
-
-            fyle_connector = FyleConnector(fyle_credentials.refresh_token, workspace_id)
+            updated_at_fyle_tpa = []
```
Comment (ashwin1111) [inline apps/fyle/tasks.py:87]: not needed

File: apps/fyle/tasks.py [hunk 9]
```diff
@@ -75,22 +84,20 @@ def async_create_expense_groups(workspace_id: int, fund_source: List[str], task_
...
+            if last_synced_at:
+                updated_at_fyle_tpa.append(
```
Comment (ashwin1111) [inline apps/fyle/tasks.py:89]: not needed

File: requirements.txt [hunk 10]
```diff
@@ -15,8 +15,9 @@ django-rest-framework==0.1.0
 djangorestframework==3.11.2
 enum34==1.1.10
 future==0.18.2
-fyle==0.9.0
+fyle==0.21.0
```
Comment (ashwin1111) [inline requirements.txt:18]: why is this needed? connector library takes care of this

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/fyle/models.py (hunk 7 above)
Comment (Sravanksk) [reply apps/fyle/models.py:108]: not needed, removing

File: requirements.txt (hunk 10 above)
Comment (Sravanksk) [reply requirements.txt:18]: yes, not needed, removing

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**